src/SwedbankPay.Sdk/Extensions/HttpClientExtensions.cs

```diff
@@ -0,0 +1,100 @@
+                    apiResponse,
+                    JsonConvert.DeserializeObject<ProblemResponse>(responseString),
+                    BuildErrorMessage(responseString, uri, apiResponse));
+
+            
+            return JsonConvert.DeserializeObject<T>(responseString, JsonSerialization.JsonSerialization.Settings);
+        }
+
+        internal static async Task<T> SendAndProcessAsync<T>(this HttpClient httpClient, HttpMethod httpMethod, Uri uri, object payload)
+            where T : class
+        {
+            var httpRequestMessage = new HttpRequestMessage(httpMethod, uri);
+
+            if (payload != null)
+            {
+                var content = JsonConvert.SerializeObject(payload, JsonSerialization.JsonSerialization.Settings);
+                httpRequestMessage.Content = new StringContent(content, Encoding.UTF8, "application/json");
+            }
+
+            var httpResponse = await httpClient.SendAsync(httpRequestMessage);
+
+            try
+            {
+                var httpResponseBody = await httpResponse.Content.ReadAsStringAsync();
+                if (!httpResponse.IsSuccessStatusCode)
+                    throw new HttpResponseException(
+                        httpResponse,
+                        JsonConvert.DeserializeObject<ProblemResponse>(httpResponseBody),
+                        BuildErrorMessage(httpResponseBody, httpRequestMessage, httpResponse));
+                return JsonConvert.DeserializeObject<T>(httpResponseBody, JsonSerialization.JsonSerialization.Settings);
+            }
+            catch (HttpResponseException ex)
+            {
+                var httpResponseBody = await httpResponse.Content.ReadAsStringAsync();
+                throw new HttpResponseException(
+                    httpResponse,
+                    message: BuildErrorMessage(httpResponseBody, httpRequestMessage, httpResponse),
+                    innerException: ex);
+            }
+        }
+
+
+        private static string BuildErrorMessage(string httpResponseBody, HttpRequestMessage httpRequest, HttpResponseMessage httpResponse)
+        {
+            return
+                $"{httpRequest.Method}: {httpRequest.RequestUri} failed with error code {httpResponse.StatusCode} using bearer token {httpRequest.Headers.Authorization.Parameter}. Response body: {httpResponseBody}";
+        }
+
+        private static string BuildErrorMessage(string httpResponseBody, Uri uri, HttpResponseMessage httpResponse)
+        {
+            return
+                $"GET: {uri} failed with error code {httpResponse.StatusCode}. Response body: {httpResponseBody}";
+        }
+
+        public static Task<T> PostAsJsonAsync<T>(this HttpClient httpClient, Uri uri, object payload)
+            where T : class
+        {
+            return httpClient.SendAndProcessAsync<T>(HttpMethod.Post, uri, payload);
+        }
+
+
+        public static Task<T> PutAsJsonAsync<P, T>(this HttpClient httpClient, Uri uri, object payload)
```

Comment:
What is P in <P, T> ?

Reply:
A forgotten leftover from a previous experiment where the syntax used was:
`public static Task<T> PutAsJsonAsync<P, T>(this HttpClient httpClient, Uri uri, P payload)`, but this made using the method more cumbersome than it was worth.